releases.md

```diff
@@ -0,0 +1,51 @@
+# Release Process
+
+This process is a work-in-progress until the first major release of the project.
+
+The process will utimately aim to have:
+
+- A published release date for the first caikit major release, upon the availability of the first beta release
+- A well-known cyclical release dates for caikit minor and patch releases, upon the availability of the first caikit major release
+- Minor releases backwards compatible with respect to the API, upon the availability of the first caikit major release
+
+## Semantic versioning
+
+caikit versions are expressed as x.y.z, where x is the major version, y is the minor version, and z is the patch version, following [Semantic Versioning](https://semver.org/spec/v2.0.0.html) terminology.
+
+## Patch releases
+
+Patch releases provide users with bug fixes and security fixes. They do not contain new features.
+
+Patch releases should be done every second week on a Wednesday (assuming there are changes since the last release). A patch release to fix a high priority regression or security issue does not have to follow this schedule, but it is highly desirable that it is released on a Wednesday if possible.
+
+### Cancelling a patch release
+
+A patch release should be cancelled:
+
+- If it falls within the week of a minor release
```

Comment:
The idea here is when we reach a **stable major release**, to have scheduled patch and minor releases. This gives prior notice to the community when to expect releases.

Therefore, for patch/minor releases milestones are created and issues/PRs are assigned to the milestone. The patch release covers bug fixes and minor release covers functionality changes.

Breaking API changes are delivered in a major release.

This is adhering to semantic version principles.

Reply:
Totally agree. My point is just that sometimes defining whether a given change breaks "the API" is tough in an interpreted language like `python` since technically _everything_ is accessible to users and therefore "part of the API" under some definition of API. I think we should (at least internally) have a plan for what to do if/when we make the decision that a change does NOT break the API, release a `patch` or `minor` change, then discover that someone's usecase was actually effected and our assessment was wrong about the nature of the change (this happens regularly in other OSS python projects, and I don't think we can guarantee that it won't happen to us)